File: src/main/ng/src/app/main/content/model/operatingParameterPLF.ts
export class OperatingParameterPLFModel {

    id: string;
    serialNumber: number;
    year: number;
    remarks: string;
    actualYearlyAveragePlfCuf: string;

    /**
     * constructor()
     * @param _operatingParameter 
     */
    constructor(_operatingParameter: any)
    {
        this.id = _operatingParameter.id || '';
        this.serialNumber = _operatingParameter.serialNumber || 0;
        this.year = _operatingParameter.year || 0;
        this.remarks = _operatingParameter.remarks || '';
        this.actualYearlyAveragePlfCuf = _operatingParameter.actualYearlyAveragePlfCuf || '';
    }
}

File: src/main/java/pfs/lms/enquiry/monitoring/operatingparameters/OperatingParameterPLF.java
package pfs.lms.enquiry.monitoring.operatingparameters;

import lombok.*;
import pfs.lms.enquiry.domain.AbstractEntity;
import pfs.lms.enquiry.monitoring.domain.LoanMonitor;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.ManyToOne;

/**
 * Created by sajeev on 16-Apr-21.
 */
@Entity
@Setter
@Getter
@ToString
@EqualsAndHashCode
@NoArgsConstructor
public class OperatingParameterPLF extends AbstractEntity implements Cloneable {

    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private LoanMonitor loanMonitor;

    private Integer serialNumber;
    private Integer year;
    private Double actualYearlyAveragePlfCuf;

    private String remarks;

    public Object clone () throws CloneNotSupportedException {
        return super.clone();
    }
 }
